tests/testmultistream.nim

```diff
@@ -0,0 +1,286 @@
+import unittest, strutils, sequtils, sugar
+import chronos
+import ../libp2p/connection, ../libp2p/multistream,
+       ../libp2p/stream, ../libp2p/connection, 
+       ../libp2p/multiaddress, ../libp2p/transport, 
+       ../libp2p/tcptransport, ../libp2p/protocol,
+       ../libp2p/crypto/crypto, ../libp2p/peerinfo,
+       ../libp2p/peer
+
+## Mock stream for select test
+type
+  TestSelectStream = ref object of LPStream
+    step*: int
+
+method readExactly*(s: TestSelectStream,
+                    pbytes: pointer,
+                    nbytes: int): Future[void] {.async.} =
+  case s.step:
+    of 1:
+      var buf = newSeq[byte](1)
+      buf[0] = 19
+      copyMem(cast[pointer](cast[uint](pbytes)), addr buf[0], buf.len())
```

Comment:
although it is only a test, but I'm curious why you cast `pbytes` to uint an then back to pointer again? isn't using `pbytes` only without any cast will serve the same purpose?

Reply:
Great question. Again, mostly stripped from chronos/libp2p code - is this cast necessary @cheatfate? I would expect `pointer` to be consumable without the cast.

Reply:
Indeed this was a pointless cast! Thanks @jangko